cmd/opampsupervisor/specification/README.md

```diff
@@ -473,6 +477,149 @@ to allow this.
 actively periodically query the health of the Collector or we can rely
 on opamp extension to report the health when it changes?*
 
+## Collector Executable Updates Flow
+
+The Supervisor will download Collector package updates when offered so
+by the OpAMP Server, if the AcceptsPackages capability is enabled.
+
+### Overview
+![Supervisor architecture diagram](agent-upgrades-e2e.png)
+
+1. During the release workflow, [Cosign] is used to sign the build artifacts.
+2. Cosign gets a signed certificate from [Fulcio]. Fulcio verifies the
+Open ID Connect token of the signer before signing the certificate.
+3. Cosign writes the computed signature and certificate to the public 
+[Rekor] instance.
+4. The release workflow adds both the release artifacts and cosign generated signature
+and certificate files to the release.
+5. An OpAMP server is made aware of the release through some mechanism
+(e.g. manually through user intervention, through polling the GitHub API,
+etc)
+6. The OpAMP server sends a PackagesAvailable message with the new agent
+release specified. See the [Expected PackagesAvailable Format](#expected-packagesavailable-format)
+section for more information.
+7. The supervisor, on startup, has retrieved the sigstore root certificates.
+8. The supervisor downloads the agent binary from the releases, using
+the URL that the OpAMP server gave it for downloading. After
+downloading, it verifies the content hash matches the server
+provided content hash from the PackagesAvailable message.
+9. The supervisor verifies the certificate via the root certificates downloaded in step 7.
```

Comment:
Can you clarify what is the difference between "verifies the certificate via the root certificates" and checks against the Rekor transparency log to verify the certificate hasn't been tampered with". I assume certificate is signed by a root certificate (the CA), so if it was tampered it would not pass the first verification step. How is possible that it passes the first check but fails the second? Are we thinking about a scenario that the root certificate is tampered with?